File: src/main/java/io/zeitwert/ddd/doc/model/base/DocRepositoryBase.java

package io.zeitwert.ddd.doc.model.base;

import org.jooq.TableRecord;

import io.zeitwert.ddd.aggregate.model.AggregateRepository;
import io.zeitwert.ddd.aggregate.model.base.AggregateRepositoryBase;
import io.zeitwert.ddd.app.service.api.AppContext;
import io.zeitwert.ddd.doc.model.Doc;
import io.zeitwert.ddd.doc.model.DocRepository;

public abstract class DocRepositoryBase<D extends Doc, V extends TableRecord<?>>
		extends AggregateRepositoryBase<D, V>
		implements DocRepository<D, V> {

	protected DocRepositoryBase(
			Class<? extends AggregateRepository<D, V>> repoIntfClass,
			Class<? extends Doc> intfClass,
			Class<? extends Doc> baseClass,
			String aggregateTypeId,
			AppContext appContext) {
		super(repoIntfClass, intfClass, baseClass, aggregateTypeId, appContext);
	}

	@Override
	public void registerPartRepositories() {
		this.addPartRepository(DocRepository.getTransitionRepository());
	}

}

File: fm-dddrive/src/main/java/io/dddrive/doc/model/base/DocRepositoryBase.kt
package io.dddrive.doc.model.base

import io.dddrive.ddd.model.base.AggregateRepositoryBase
import io.dddrive.doc.model.Doc
import io.dddrive.doc.model.DocPartTransition
import io.dddrive.doc.model.DocRepository
import io.dddrive.doc.model.impl.DocPartTransitionImpl

abstract class DocRepositoryBase<D : Doc>(
	intfClass: Class<out Doc>,
	aggregateTypeId: String,
) : AggregateRepositoryBase<D>(intfClass, aggregateTypeId),
	DocRepository<D> {

	override fun doLogChange(property: String): Boolean {
		if (NotLoggedProperties.contains(property)) {
			return false
		}
		return super.doLogChange(property)
	}

	override fun registerParts() {
		this.addPart(DocPartTransition::class.java, ::DocPartTransitionImpl)
	}

	companion object {

		private val NotLoggedProperties = setOf("caseDef", "isInWork", "transitionList")
	}

}
